cpp/src/arrow/acero/hash_join_node_test.cc

```diff
@@ -1366,7 +1368,12 @@ void TestHashJoinDictionaryHelper(
                        TableFromExecBatches(output_schema, expected_batches));
 
   // Compare results
-  AssertTablesEqualIgnoringOrder(expected, output);
+  if (ignore_order) {
```

Comment:
This is needed because there is no way to sort lists as of now.

Reply:
can you make this an in-code comment? it seems useful to comment this